cpp/utils.cpp

```diff
@@ -207,6 +207,59 @@ double dolfinx_contact::dR_minus(double x) { return double(x < 0); }
 //-------------------------------------------------------------------------------------
 
 double dolfinx_contact::dR_plus(double x) { return double(x > 0); }
+
+std::array<double, 3> dolfinx_contact::ball_projection(std::array<double, 3> x,
+                                                       double alpha,
+                                                       std::size_t bs)
+{
+  // Compute norm of vector
+  double norm = 0;
+  for (std::size_t j = 0; j < bs; ++j)
+    norm += x[j] * x[j];
+  norm = std::sqrt(norm);
+  std::array<double, 3> proj = {0, 0, 0};
+
+  // If x inside ball return x
+  if (norm <= alpha)
+    for (std::size_t j = 0; j < bs; ++j)
+      proj[j] = x[j];
```

Comment:
Same comment as above, could use
```suggestion
    std::copy(x.cbegin(), x.cend(), proj.begin());
```
(please test this locally before applying the comment, as I'm prone typos xD